### DISCUSSION TITLE
Erro "Partial method must have an implementation part because it has accessibility modifiers" ao utilizar Source Generator para Regex

### DISCUSSION THREAD

**Discussion by RdoEgito** [balta-io/forum#237] [Tira Dúvidas] [answered]
Estou realizando o curso "Aplicando Orientação a Objetos em Projetos Reais com C# 11 e .NET 7" e, na aula "Source Generators", a chamada do método `IsMatch()` da classe `Regex` é substituída pelo partial method `UrlRegex()`. Porém, quando eu replico a modificação no meu código, eu recebo o erro "Partial method must have an implementation part because it has accessibility modifiers", conforme a imagem abaixo.

![image](https://user-images.githubusercontent.com/33696073/214069345-949238ef-ef55-46ca-86a1-bd207ef4fb90.png)

Estou utilizando o .Net 7.0 na minha aplicação:
![image](https://user-images.githubusercontent.com/33696073/214069915-2bc2136a-3114-4fc0-b320-93599bb282ad.png)

Existe algo que eu devo fazer para que o Source Generator funcione corretamente?

Grato desde já pelo suporte.

**Comment by andrebaltieri** [comment, marked as answer]
@RdoEgito como vai? Sua classe está marcada como `partial` também?
É necessário...

Veja o exemplo:
https://github.com/balta-io/2818/blob/main/src/UtmBuilder/UtmBuilder.Core/ValueObjects/Exceptions/InvalidUrlException.cs

**Reply by RdoEgito** [reply]
@andrebaltieri está sim. Inclusive, copiei o código exatamente como está aí para garantir que não houvesse erro e, ainda assim, recebo o mesmo erro.

Pode ter a ver com a versão do Visual Studio? Estou a versão 17.3.6, mas como o PC é do trabalho, ele ainda não permite instalação de uma versão mais recente que essa.

**Reply by andrebaltieri** [reply]
Hum, esta versão do Visual Studio era para suportar... seu projeto está com .NET 7?

**Reply by RdoEgito** [reply]
Sim, está com .NET 7. Abaixo está o conteúdo do arquivo UtmBuilder.Core.csproj:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```

Eu conferi e o .NET está devidamente instalado:
![image](https://user-images.githubusercontent.com/33696073/214376433-a043f7ed-12ae-4d40-bfe3-3b9fdb892274.png)

**Reply by andrebaltieri** [reply]
Que estranho, o meu rodou até no Mac... será que tem algo bloqueando a execução do processo do .NET na sua máquina? No Event Viewer não loga nada?

Só contextualizando, neste caso, quando você cria um código com base no source generator, o .NET (+IDE) rodam um processo que gera um arquivo (outra partial). No seu caso, o erro indica que esta partial não existe.

Pode tentar daqr um clean no projeto e buildar novamente? As vezes ficou algum lixo no bin/obj.

**Reply by RdoEgito** [reply]
Finalmente consegui atualizar o meu Visual Studio para a versão 17.4.4 e o problema foi resolvido. Inclusive, o próprio IntelliSense agora sugere a criação do método partial quando utilizo a classe Regex, assim como foi mostrado na aula.

**Reply by andrebaltieri** [reply]
@RdoEgito aeeeeee 🚀
